monai/bundle/scripts.py

```diff
@@ -982,41 +1092,136 @@ def ckpt_export(
+        config_file: filepath of the config file to save in the TensorRT based torchscript model and extract network
+            information, the saved key in the model is the config filename without extension, and the saved config
+            value is always serialized in JSON format no matter the original file format is JSON or YAML.
+            it can be a single file or a list of files. if `None`, must be provided in `args_file`.
+        key_in_ckpt: for nested checkpoint like `{"model": XXX, "optimizer": XXX, ...}`, specify the key of model
+            weights. if not nested checkpoint, no need to set.
+        precision: the weight precision of converted TensorRT engine based torchscript models. Should be 'fp32' or 'fp16'.
+        inputs_shape: the inputs' shape that is used to generate random input during the convert. Should be a list with elements
+            like [[N1, C1, H1, W1], ... ,[Nk, Ck, Hk, Wk]] or [[N1, C1, H1, W1, D1], ...,[Nk, Ck, Hk, Wk, Dk]] for k inputs.
+        ir: the intermediate representation way to transform a pytorch module to a TensorRT engine based torchscript. Could
+            be choose from `(script, trace)`.
+        dynamic_batchsize: a list contains range lists with three elements to define the batch size range of inputs for
+            the converted model. In each element, the three number means `MIN_BATCH, OPT_BATCH, MAX_BATCH` in order.
+            Please notice that if you have m inputs that need dynamic_batchsize, you should input m range lists like
+            [[MIN_BATCH_1, OPT_BATCH_1, MAX_BATCH_1], ..., [MIN_BATCH_m, OPT_BATCH_m, MAX_BATCH_m]] corresponding to the
+            shape in the `inputs_shape` parameter.
+        args_file: a JSON or YAML file to provide default values for `meta_file`, `config_file`,
+            `net_id` and override pairs. so that the command line inputs can be simplified.
+        override: id-value pairs to override or add the corresponding config content.
+            e.g. ``--_meta#network_data_format#inputs#image#num_channels 3``.
+
+    """
+    _args = _update_args(
+        args=args_file,
+        net_id=net_id,
+        filepath=filepath,
+        meta_file=meta_file,
+        config_file=config_file,
+        ckpt_file=ckpt_file,
+        key_in_ckpt=key_in_ckpt,
+        precision=precision,
+        inputs_shape=inputs_shape,
+        ir=ir,
+        dynamic_batchsize=dynamic_batchsize,
+        **override,
+    )
+    _log_input_summary(tag="trt_export", args=_args)
+    (
+        filepath_,
+        ckpt_file_,
+        config_file_,
+        net_id_,
+        meta_file_,
+        key_in_ckpt_,
+        precision_,
+        inputs_shape_,
+        dynamic_batchsize_,
+        ir_,
+    ) = _pop_args(
+        _args,
+        "filepath",
+        "ckpt_file",
+        "config_file",
+        net_id="",
+        meta_file=None,
+        key_in_ckpt="",
+        precision="fp32",
+        inputs_shape=[],
+        dynamic_batchsize=[[1, 4, 8]],
```

Comment:
Why set default value for this arg?

Thanks.

Reply:
If users don't specifiy the `inputs_shape` parameter, the default batchsize `inputs_shape` parsed from the `metadata.json` would be 1, which doesn't support slidewindow inference. So by default, I try to make the dynamic batchsize range to be [1, 8] to fit some slidewindow cases.

Reply:
It looks very hacky, someday we may see a bug: for unknown reason, some sliding window can work, some sliding window with batchsize > 8 can't work.. just because you potentially set a [1, 8] batchsize here. I would suggest to leave it for users to explicitly set.
What do you think?

Thanks.

Reply:
Ok, I can update this parameter.